Moving mapinfo toggle cvars into the HUD settings

comatoad opened the pull request:
Closes #956 

<!-- As the issue described, small amount of code to move mapinfo into it's own settings panel in the HUD settings. -->

### Checklist
- [x] **I have thoroughly tested all of the code I have modified/added/removed to ensure something else did not break**
 -sidenote: Some features I tested in an unorthodox matter e.g. displaying mapinfo authors wouldn't display (probably because it hooks into momentum's online component that I don't have access to), but toggled just fine according to the dev console.

- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] If I have added or modified any visual assets (models, materials, panels, effects, etc), I have taken screenshots / videos of them and attached them to this PR directly (screenshots uploaded through github, videos uploaded to youtube and linked)
- [x] If I have modified any console command, console variable, or momentum entity, I have opened an issue (or a PR) for it in the [Momentum Mod documentation repository](https://github.com/momentum-mod/docs)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
-sidenote: understanding momentum took me a while, so if my commit history is too ridiculous to follow - let me know and I'll make it more legible.
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

Edit: As requested, visual documentation of the panel has been included -- because I modified the height of the HUDPage, I can create a video to document that if necessary.
<!-- If any of these items are giving you doubts, please ask about it in the Discord! -->

![hl2_9smFPfjrUO](https://user-images.githubusercontent.com/72104872/97254537-98478500-17dc-11eb-87c8-91dd45f56d8c.png)
![hl2_K8qCT6rpdu](https://user-images.githubusercontent.com/72104872/97254543-9b427580-17dc-11eb-8d8c-ecf3877c2314.png)

Gocnak commented on mp/src/creategameprojects.ps1:
```diff
@@ -8,7 +8,7 @@ if (-NOT ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdent
 # This is the path to your steamworks dev folder (depo). This should be the folder with hl2.exe inside of it.
 # Your local momentum folder will be sym linked INSIDE this folder as momentum
-$path = "E:\Steamworks\dev\Momentum"
+$path = "E:\Coding Projects\MomentumDev"
```
You do not want to commit this, and you can actually revert this file back after running `creategameprojects.bat` as admin, since the symlink is only needed to be created once. Any future runs of `creategameprojects.bat` (when you change branch or edit any `.vpc` file) can be run as normal (non-admin).

Gocnak commented on mp/game/momentum/resource/ui/settings/Settings_HUD.res:
```diff
@@ -1621,4 +1616,286 @@
+        "auto_tall_tocontents" "1"
+        "use_proportional_insets"		"0"
+        "Default"		"0"
+    }
```
Looks like we're missing an extra `}` here to close off the original keyvalue:

```suggestion
    }
}
```

Gocnak commented on mp/game/momentum/resource/momentum_english_ref.res:
```diff
@@ -493,10 +493,16 @@
+        "MOM_MapInfo_Label" "Map info"
+        "MOM_Settings_Show_MapName" "Display map name"
+        "MOM_Settings_Show_MapAuthor" "Display map author"
+        "MOM_Settings_Show_MapDifficulty" "Display map difficulty"
+        "MOM_Settings_Show_MapStatus" "Show map status"
 
         // Replay Info
         // Spec GUI
-        "MOM_WatchingReplay"    "REPLAY"
+        "WatchingReplay"    "REPLAY"
```
Accidental change?

Gocnak reviewed (changes requested):
Looking good but it seems like it's slightly out of date! You need to do a couple things:

1. You can remove the `surf_lt_omnific.nav` file. I thought those were gitignored actually, so feel free to add `*.nav` to the `.gitignore`, [around where the momentum ignore stuff is](https://github.com/momentum-mod/game/blob/develop/.gitignore#L73-L80).

2. You're going to want to rebase onto our develop, which is going to cause a merge conflict but it's not so bad! You'll need to use the git command line, which if you're using GitHub desktop you can access by pressing CTRL + \`. (It's under Repository -> Open in Command Line)

First, make sure you have our repo added as a remote:
`git remote add upstream https://github.com/momentum-mod/game.git`

Then you update our copy:
`git fetch -p upstream`

Make sure that you're on your main branch:
`git checkout develop`

Then update your branch with our stuff:
`git rebase upstream/develop`

With your develop going under rebase, you're going to hit a conflict on the `momentum_english.txt` file. Don't worry! This sucker is annoying like that and happens whenever there's parallel work done in localization stuffs. We'll probably get a better way to handle this in the future, but for now, in this "merge conflict" state:

Re-run `tokenizer.py` to generate an up-to-date localization file.
Then, back in your command line:
```
git add .
git rebase --continue
```

It *should* continue as normal, rebasing your develop on top of ours! Then all you need to do is, and *ONLY DO THIS IF IT WORKED!* Meaning, you do `git status` and there's no stashed changes or anything:

`git push -f`

This force pushes your repo and overwrites the old history (your old develop branch) with the new rebased history. Best of all, no merge commits!

It should automatically show up here when you do that. Try to save it for last, after you acknowledge all of my requested changes. Any questions, feel free to ask me on Discord! 😄

Gocnak commented:
Also, you modified some visual assets (panel), as is checkbox 4, so please take a screenshot of what the new settings look like and attach it to the original PR message in an edit :)

comatoad commented:
Thank you for the cohesive feedback! I'll make the adequate changes and get back to you as soon as possible.

Gocnak commented:
Do note that the longer you take, the more problematic the rebase may become!

comatoad commented:
Hello again - I believe I have fixed all the issues you have mentioned, sorry for the delay. I wasn't familiar with Github Desktop, so I appreciate the guide for rebasing 👍 .

If there are any other silly issues I made in this branch, I'll fix them ASAP.

Gocnak reviewed (changes requested):
A couple of things noticed, you updated by merging (hence the merge commit) and you should try my rebase steps again. You can see it's messed up with the extra commits already on our repo and the number of files changed.

Another thing I noticed in the photo: are the controls properly centered? Can you take a full screenshot with the other settings on screen too so that I can compare them?

comatoad commented:
Oops, sorry about that.  I have fixed the rebasing issue.
I moved the Map Info settings panel above the comparisons panel for a cleaner looking UI - as far as I can tell, it's centered, but it definitely wasn't earlier.  The photos are in the original post now - if you need any further documentation or fixes let me know

Gocnak commented on mp/game/momentum/resource/ui/settings/Settings_HUD.res:
```diff
@@ -683,11 +683,6 @@
-    // ================================================================================
-    // ================================================================================
-    // ================================================================================
-    // ================================================================================
```
These are used to signify separate sections, was this accidentally removed?

Gocnak reviewed (approved):
I guess removing that comment doesn't super matter. Good work, PM me on Discord when you get the chance to.
